src/core/features/search/components/global-search-result/global-search-result.ts

```diff
@@ -31,9 +34,25 @@ export class CoreSearchGlobalSearchResultComponent implements OnChanges {
      * @inheritdoc
      */
     ngOnChanges(): void {
+        this.renderedContext = this.computeRenderedContext();
         this.renderedIcon = this.computeRenderedIcon();
     }
 
+    /**
+     * Calculate the value of the context to render.
+     *
+     * @returns Rendered context.
+     */
+    private computeRenderedContext(): CoreSearchGlobalSearchResultContext | null {
+        const context = this.result.context ?? {};
```

Comment:
```suggestion
        const context = { ...this.result.context };
```
IMO it's better to create a copy of the object here, because you're deleting a property later (courseName) so your code modifies the input object and this can affect the outer components.